File: content/problems/2364-count-number-of-bad-pairs/solutions/difference-key-frequency/Solution.kt
class Solution {
    fun countBadPairs(nums:IntArray):Long{
        val frequency=HashMap<Int,Int>();
        var answer=0L;
        for(index in nums.indices){
            val key=nums[index]-index;
            answer+=index-(frequency[key]?:0);
            frequency[key]=(frequency[key]?:0)+1
        };
        return answer
    }
}

File: content/problems/2364-count-number-of-bad-pairs/solutions/difference-key-frequency/solution.cpp
class Solution {
    public:long long countBadPairs(vector<int>& nums){
        unordered_map<int,int> frequency;
        long long answer=0;
        for(int index=0;index<(int)nums.size();++index){
            int key=nums[index]-index;
            answer+=index-frequency[key];
            ++frequency[key];
        }
        return answer;
    }
};
